File: backend-roadstar/app/Models/Vehicle.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Vehicle extends Model
{
    use HasFactory;

    protected $fillable = [
        'brand',
        'model',
        'year',
        'license_plate',
        'color',
        'mileage',
        'fuel_type',
        'transmission',
        'seats',
        'daily_rate',
        'status',
        'is_featured',
        'image',
        'description',
        'features',
        'category',
    ];

    protected $casts = [
        'year' => 'integer',
        'mileage' => 'integer',
        'seats' => 'integer',
        'daily_rate' => 'decimal:2',
        'is_featured' => 'boolean',
        'features' => 'array', // JSON cast
    ];

    protected $appends = ['name', 'price', 'passengers', 'fuel', 'featured', 'available', 'type'];

    public function getNameAttribute()
    {
        return trim($this->brand . ' ' . $this->model);
    }

    public function getPriceAttribute()
    {
        return number_format($this->daily_rate, 0, '.', ' ') . ' FCFA';
    }

    public function getPassengersAttribute()
    {
        return $this->seats;
    }

    public function getFuelAttribute()
    {
        return $this->fuel_type;
    }

    public function getFeaturedAttribute()
    {
        return $this->is_featured;
    }

    public function getAvailableAttribute()
    {
        return $this->status === 'available';
    }

    public function getTypeAttribute()
    {
        return $this->category ?? 'Véhicule';
    }
}

File: mobile-roadstar/lib/features/vehicles/domain/entities/vehicle.dart
import 'package:equatable/equatable.dart';

class Vehicle extends Equatable {
  final int id;
  final String brand;
  final String model;
  final int year;
  final String licensePlate;
  final String color;
  final int mileage;
  final String fuelType;
  final String transmission;
  final int seats;
  final double dailyRate;
  final String status;
  final bool isFeatured;
  final String? description;
  final String? category;
  final String? image;

  const Vehicle({
    required this.id,
    required this.brand,
    required this.model,
    required this.year,
    required this.licensePlate,
    required this.color,
    required this.mileage,
    required this.fuelType,
    required this.transmission,
    required this.seats,
    required this.dailyRate,
    required this.status,
    required this.isFeatured,
    this.description,
    this.category,
    this.image,
  });

  @override
  List<Object?> get props => [
        id,
        brand,
        model,
        year,
        licensePlate,
        color,
        mileage,
        fuelType,
        transmission,
        seats,
        dailyRate,
        status,
        isFeatured,
        description,
        category,
        image,
      ];
}
